src/worker/core/trade/summary.ts

```diff
@@ -101,9 +103,13 @@ const summary = async (teams: TradeTeams): Promise<TradeSummary> => {
 			}
 		}),
 	);
+	if (g.get("tradeMatchingPercentage") === 0) {
```

Comment:
If you return here, then the `hardCapCondition` code below never runs. To fix that, add `&& !g.get("hardCap")` to this condition.